### PR TITLE
Fixed UI issue related to cropping of keyboard and overlapping of keyboard with text box

### LINKED ISSUE #68: [onebillion]: Fix the overlap issue of Text Box with Keyboard in Play Zone section

### PR DESCRIPTION
**Pull request by piyushonkar**
|PR checklist|Status|
|-----|------|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes| 
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA|
|Is code commenting done properly for the changes made?|Yes|
|Is correct variable naming convention followed?|Yes|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|No|
|Are the message strings used in the application taken from strings.xml file?|NA|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|NA|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|NA|
|Are updated images compressed (< 100 KB)?|NA|
|Typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|No| 

Fixed #68

### CODE REVIEW

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java [hunk 1]
```diff
@@ -131,12 +124,15 @@ public void prepare()
...
             textScale = Float.valueOf(p);
 
-        String[] themea = ((String)eventAttributes.get("themes")).split(",");;
+        String[] themea = ((String)eventAttributes.get("themes")).split(",");
         themeNames = Arrays.asList(themea);
 
-        textBox = new OBScrollingText(objectDict.get("text_box").frame());
-        textBox.setZPosition(5);
-        textBox.setFillColor(Color.WHITE);
+        float textBottom = objectDict.get("keyboard_rect").frame.top;
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: frame().top

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java [hunk 2]
```diff
@@ -379,6 +377,15 @@ public void loadKeyboard(List<List<String>> letters)
         float height = keyNormal.height();
         float minLeft = keyboardRect.left() + buttonDistance;
         float maxRight = minLeft;
+        WindowManager wm = (WindowManager) activity.getSystemService(Context.WINDOW_SERVICE);
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: Use a better variable name

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java [hunk 3]
```diff
@@ -379,6 +377,15 @@ public void loadKeyboard(List<List<String>> letters)
...
+        Point size = new Point();
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: Even for this

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java [hunk 4]
```diff
@@ -379,6 +377,15 @@ public void loadKeyboard(List<List<String>> letters)
...
+        wm.getDefaultDisplay().getRealSize(size);
+        float screenWidth = (float) size.x, screenHeight = (float) size.y;
+        if(screenHeight - 20f < keyboardRect.frame.bottom )
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: Can we avoid hard-coding of values?

**Review by akashdabhi03 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java (hunk 2 above)
Comment (piyushonkar) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: Done

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java (hunk 4 above)
Comment (piyushonkar) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: Done

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java (hunk 3 above)
Comment (piyushonkar) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java]: Done.

**Review by akashdabhi03 — APPROVED** [review]

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java [hunk 5]
```diff
@@ -366,79 +330,74 @@ public void loadTheme(String name)
...
         float maxBottom = minTop;
         float top = minTop;
         float height = keyNormal.height();
         float minLeft = keyboardRect.left() + buttonDistance;
         float maxRight = minLeft;
+        WindowManager windowManager = (WindowManager) activity.getSystemService(Context.WINDOW_SERVICE);
+        Point sizeScreen = new Point();
+        windowManager.getDefaultDisplay().getRealSize(sizeScreen);
+        float screenWidth = (float) sizeScreen.x, screenHeight = (float) sizeScreen.y;
+        // Check for bottom of keyboard not exceeding screen size, if it exceeds then set bottom of keyboard
```
Comment (mathewnik90) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/oc_playzone/OC_PlayZoneTypewrite.java:350]: Check if bottom of keyboard exceeds screen size, if it does then set bottom

**Pull request merged**